src/components/SelectOption/SelectOption.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import usersFromServer from '../../api/users';
+
+interface Props {
+  name: string;
+  value?: string;
+  label: string;
+  placeholder: string;
+  hasError?: boolean;
+  onSelectedUser: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const SelectOption: React.FC<Props> = ({
+  label,
+  placeholder,
+  hasError = false,
+  onSelectedUser,
+}) => {
+  return (
+    <div className="field">
+      <label className="label" htmlFor="title">
+        {label}
+      </label>
+      <select
+        data-cy="userSelect"
+        defaultValue="0"
+        onChange={event => onSelectedUser(Number(event.target.value))}
+      >
+        <option value="0" disabled>
+          {placeholder}
+        </option>
+
+        {usersFromServer.map((user, index) => {
+          return (
+            <option key={user.id} value={index + 1}>
```

Comment:
[CHECKLIST ITEM 2] - Don't repeat yourself. The value 'index + 1' is derived from the map index and may not be reliable as the user's identifier. Use the 'user.id' directly as the value for the option element.